packages/sanity/src/core/tasks/components/form/tasksFormBuilder/FormCreate.tsx

```diff
@@ -47,7 +44,52 @@ export function FormCreate(props: ObjectInputProps) {
   }, [setViewMode])
   const {handleRemove, removeStatus} = useRemoveTask({id: value._id, onRemoved: onRemove})
   const {t} = useTranslation(tasksLocaleNamespace)
-  const handleCreate = useCallback(() => {
+  const {data} = useTasks()
+  const savedTask = data.find((task) => task._id === value._id)
+
+  useEffect(() => {
+    // This useEffect takes care of closing the form when a task entered the "creation" state.
+    // That action is async and we don't have access to the promise, once the value is updated in the form we will close the form.
+    if (creating && savedTask?.createdByUser) {
+      telemetry.log(TaskCreated)
+      toast.push({
+        closable: true,
+        status: 'success',
+        title: t('form.status.success'),
+      })
+
+      setCreating(false)
+      if (createMore) {
+        setViewMode({type: 'create'})
+      } else {
+        setActiveTab('subscribed')
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [creating, savedTask?.createdByUser])
+
+  useEffect(() => {
+    // If after 10 seconds the task is still in the "creating" state, show an error and reset the creating state.
+    let timeoutId: NodeJS.Timeout | null = null
+    if (creating) {
+      timeoutId = setTimeout(() => {
+        setCreating(false)
+        toast.push({
+          closable: true,
+          status: 'error',
+          title: t('form.status.error.creation-failed'),
+        })
+      }, 10000)
+    }
+    // Cleanup function to clear the timeout
+    return () => {
+      if (timeoutId) clearTimeout(timeoutId)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [creating])
```

Comment:
Same here, lets bake this into one `useDocumentOperationEvent` handler if possible.

Reply:
If possible, I think this would be more robust. Note that `useDocumentOperationEvent` will also trigger for work done by collaborators (might not be an issue here)

Reply:
Just tried with `useDocumentOperation` event and it returns an object like this:
```
{
    "type": "success",
    "op": "commit",
    "id": "0a6836ee-ec1d-4f87-b1c7-bd7d3b668e5d"
}
```

It will return the same object for every operation that is done by the form, so if you are typing or changing any value it will continue showing the same values. 
Would be great if there is a way to handle this, is there any option to get access to the transaction that is executed on that operation?

Reply:
Good point. Transaction IDs are generated as the last step before mutations are submitted to the backend, so there's no way of knowing what they will be at the time the operation is executed.